docs/proposals/automated-node-removal.md

```diff
@@ -0,0 +1,72 @@
+# Proposal: Automation of node removal procedure
+
+Last updated: 24.06.2021
+
+## Abstract
+
+Implementation of the option to delete CSI CRs from a node, which is going to be removed. 
+
+## Background
+
+User manual actions for node removal are described in [doc](https://github.com/dell/csi-baremetal/blob/master/docs/node-removal.md).
+The goal of this proposal is automating most of the steps of that procedure.
+
+## Proposal
+
+#### User API
+
+CSI CRs deletion(Volumes, AC, ACR, Drives, LVGs, Csibmnode) may be triggerred on taint a Node:
+- `node.dell.com/drain=drain:NoSchedule` - removal (node should be deleted from the cluster)
+- `node.dell.com/drain=planned-downtime:NoSchedule` - maintenance (node may be restored with the same hostname/ip)
+
+## Implementation
+
+#### Removal
+
+1. CSI Operator must reconcile Nodes and check `node.dell.com/drain=drain:NoSchedule` taint
+2. Label Csibmnode with `node-should-be-removed`
+3. Check that there are no PV for the Node (reconcile Csibmnode with `node-should-be-removed`) [ISSUE-1]
```

Comment:
What to do if PVs aren't removed due to some issue?

Reply:
updated